From 6232cb00f1393aa7dc0aa478341f2b32012feb31 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sat, 25 Apr 2020 14:19:35 +0200
Subject: [PATCH 3/5] Fix 'make install' on Alpine Linux

I'm not quite sure what the actual problem and why it doesn't work (musl
problem?, toolchain problem?) but here are the results from my
investigation:

* Relative and absolute LD_PRELOAD doesn't work
* Relative LD_LIBRARY_PATH doesn't work
* Absolute LD_LIBRARY_PATH *does* work

So let's use the absolute LD_LIBRARY_PATH.
---
 GSettings/gsettings-qt.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/GSettings/gsettings-qt.pro b/GSettings/gsettings-qt.pro
index 218eeea..2642981 100644
--- a/GSettings/gsettings-qt.pro
+++ b/GSettings/gsettings-qt.pro
@@ -25,6 +25,6 @@ INSTALLS += extra
 
 qmltypes.path = $$installPath
 qmltypes.files = plugins.qmltypes
-qmltypes.extra = export LD_PRELOAD=../src/libgsettings-qt.so.1; $$[QT_INSTALL_BINS]/qmlplugindump -notrelocatable GSettings 1.0 .. > $(INSTALL_ROOT)/$$installPath/plugins.qmltypes
+qmltypes.extra = export LD_LIBRARY_PATH=$$PWD/../src; $$[QT_INSTALL_BINS]/qmlplugindump -notrelocatable GSettings 1.0 .. > $(INSTALL_ROOT)/$$installPath/plugins.qmltypes
 INSTALLS += qmltypes
 
-- 
2.26.2

